### PR TITLE
Fix and waive lint errors in various modules

### PR DESCRIPTION
**Pull request by vogelpi**
This PR fixes and waives lint errors in various modules. Since I need someone to check that some of the waivers work in AscentLint, I decided to collect all commits in a single PR.

@msfschaffner , could you please check the following modules:
- aes: 3 errors are expected to remain
- rv_core_ibex: 0 errors expected to remain
- flash_ctrl: 1 error is expected to remain

### CODE REVIEW

File: hw/ip/rv_core_ibex/lint/rv_core_ibex.waiver
Comment (vogelpi) [inline hw/ip/rv_core_ibex/lint/rv_core_ibex.waiver:116]: @GregAC @msfschaffner I've reviewed the mechanism for generating the reset of the lockstep core and it looks okay to me. Our implementation follows common guidelines. I think it's safe to waive these lint errors.

Comment (msfschaffner) [reply hw/ip/rv_core_ibex/lint/rv_core_ibex.waiver:118]: The mechanisms look fine to me, the only question I would have is whether we should use `prim_clock_mux2` and `prim_flop` to implement it.
Specifically, for other scan bypasses, we are using primitives as well, e.g.:
https://github.com/lowRISC/opentitan/blob/d97b4a0fe8c4370169799cb4b2783833b716e26f/hw/ip/lc_ctrl/rtl/lc_ctrl.sv#L170-L188

**Review by msfschaffner — COMMENTED** [review]
Overall LGTM, but a few adjustments are needed to make this work (see fixup commits).

Also, I still get the following lint errors. 
Especially the one about `wdata_i` not being read from sounds a bit odd...

-----------------------------------

## AES Lint Results

### Monday January 10 2022 21:51:54 UTC
### GitHub Revision: [`b322f73da`](https://github.com/lowrisc/opentitan/tree/b322f73da2a0748e9b30e2a52b886bda16ffdd9c)<br>Foundry Revision: `8d05ac77`
### Branch: lint
### Tool: ASCENTLINT

|  Build Mode  |  Flow Warnings  |  Flow Errors  |  Lint Warnings  |  Lint Errors  |
|:------------:|:---------------:|:-------------:|:---------------:|:-------------:|
|   default    |       0 W       |      0 E      |       0 W       |      4 E      |

### Messages for Build Mode `'default'`
#### Lint Errors
```
E   NOT_READ:   aes_control.sv:133        Signal 'sec_start_trigger_delay' is not read from in module 'aes_control'                       New                            

E   NOT_READ:   aes_prng_clearing.sv:55   Signal 'sec_skip_prng_reseeding' is not read from in module 'aes_prng_clearing'                 New                            

E   NOT_READ:   aes_prng_masking.sv:82    Signal 'sec_skip_prng_reseeding' is not read from in module 'aes_prng_masking'                  New                            

E   NOT_READ:   aes_prng_masking.sv:197   Signal 'sec_allow_forcing_masks' is not read from in module 'aes_prng_masking'                  New                            

```

## FLASH_CTRL Lint Results

### Monday January 10 2022 21:51:54 UTC
### GitHub Revision: [`b322f73da`](https://github.com/lowrisc/opentitan/tree/b322f73da2a0748e9b30e2a52b886bda16ffdd9c)<br>Foundry Revision: `8d05ac77`
### Branch: lint
### Tool: ASCENTLINT

|  Build Mode  |  Flow Warnings  |  Flow Errors  |  Lint Warnings  |  Lint Errors  |
|:------------:|:---------------:|:-------------:|:---------------:|:-------------:|
|   default    |       0 W       |      0 E      |       0 W       |      1 E      |

### Messages for Build Mode `'default'`
#### Lint Errors
```
E   PARAM_NOT_USED:   prim_generic_flash_bank.sv:15   Parameter 'MetaDataWidth' not used in module 'prim_generic_flash_bank'                 New                                                                                           

```

## RV_CORE_IBEX Lint Results

### Monday January 10 2022 21:51:54 UTC
### GitHub Revision: [`b322f73da`](https://github.com/lowrisc/opentitan/tree/b322f73da2a0748e9b30e2a52b886bda16ffdd9c)<br>Foundry Revision: `8d05ac77`
### Branch: lint
### Tool: ASCENTLINT

|  Build Mode  |  Flow Warnings  |  Flow Errors  |  Lint Warnings  |  Lint Errors  |
|:------------:|:---------------:|:-------------:|:---------------:|:-------------:|
|   default    |       0 W       |      0 E      |       0 W       |      2 E      |

### Messages for Build Mode `'default'`
#### Lint Errors
```
E   HIER_BRANCH_NOT_READ:   prim_ram_1p.sv:34               Net 'wdata_i' in module 'prim_ram_1p' (Width=32'h1c,Depth=32'h100,DataBitsPerMask=32'h1c) is not read from,                                                                             New            
                                                                               

E   HIER_BRANCH_NOT_READ:   prim_ram_1p.sv:34               Connected net 'wdata_i' at prim_generic_ram_1p.sv:22 is not read from in module 'prim_generic_ram_1p' (Width=32'h1c,Depth=32'h100,DataBitsPerMask=32'h1c)   prim_generic_ram_1p.sv:22   New            
                                                                               

```

**Comment by tjaychen** [comment]
i'll take the flash one. I forgot to remove a parameter (mostly because i got lazy about changing the proprietary side) after merging the metadata and data memories.

File: hw/ip/rv_core_ibex/lint/rv_core_ibex.waiver
Comment (vogelpi) [reply hw/ip/rv_core_ibex/lint/rv_core_ibex.waiver:118]: This makes sense Michael, I've filed an Ibex PR to achieve this. See lowRISC/Ibex#1505, PTAL.

**Comment by vogelpi** [comment]
Thanks a lot @msfschaffner for taking a look and fixing the waivers!

I've now rebased this PR and also added another commit to fix the lint errors around `wdata_i` not being read from inside the abstract prim wrappers. My assumption is that this is a result of AscentLint blackboxing RAMs to speed up runtime (there are already comments around this in the `prim_generic_ram_xp` waivers) and that maybe with an updated tool version, this this propagates into the wrapper as well.

**Comment by vogelpi** [comment]
> i'll take the flash one. I forgot to remove a parameter (mostly because i got lazy about changing the proprietary side) after merging the metadata and data memories.

Thanks @tjaychen for taking care of this.

File: hw/ip/prim/util/primgen/abstract_prim.sv.tpl
Comment (msfschaffner) [inline hw/ip/prim/util/primgen/abstract_prim.sv.tpl:18]: You have to modify the corresponding `//ri lint_check_on` comment below as well - otherwise the tool gets upset and complains ;).

**Review by msfschaffner — APPROVED** [review]
LGTM, can be merged after addressing my last comment.

File: hw/ip/prim/util/primgen/abstract_prim.sv.tpl
Comment (vogelpi) [reply hw/ip/prim/util/primgen/abstract_prim.sv.tpl:18]: Ah, that's a stupid mistake, sorry. It's now fixed.

**Review by GregAC — APPROVED** [review]
Only reviewing the `rv_core_ibex` changes, happy with those.

**Pull request merged**
